src/main/scala/viper/gobra/translator/implementations/translator/StatementsImpl.scala

```diff
@@ -37,6 +39,26 @@ class StatementsImpl extends Statements {
     def goA(a: in.Assertion): CodeWriter[vpr.Exp] = ctx.ass.translate(a)(ctx)
     def goE(e: in.Expr): CodeWriter[vpr.Exp] = ctx.expr.translate(e)(ctx)
 
+    /**
+      * Translates a go call to a function or method with pre-condition `pre` which is parameterized by
+      * formal parameters `formalParams` and is instantiated with `args`
+      */
+    def translateGoCall(pre: Vector[in.Assertion],
+                        formalParams: Vector[in.Parameter.In],
+                        args: Vector[in.Expr]): Writer[vpr.Stmt] = {
+      Violation.violation(args.length == formalParams.length, "number of passed arguments must match number of expected arguments")
+      val errorT = (x: Source.Verifier.Info, _: ErrorReason) => PreconditionError(x) dueTo GoCallPreconditionError(x)
+      for {
+        vArgss <- sequence(args map goE)
+        funcArgs <- sequence(formalParams map goE)
+        substitutions = (funcArgs zip vArgss).toMap
+        preCond <- sequence(pre map goA)
+        preCondInstance = preCond.map{ _.replace(substitutions) }
+        and = vu.bigAnd(preCondInstance)(pos, info, errT)
+        _ <- assert(and, errorT)
+      } yield vpr.Exhale(and)(pos, info, errT)
```

Comment:
I think this needs to be surrounded by seqn( ... )

Reply:
Most likely I am missing something but why should I use seqn here if I only want to write a single Exhale statement?

Reply:
Without seqn you can get some reordering issues. If you have 
```
for {
  w <- for { _ <- write(B) } yield C
} yield vpr.Seqn(Vector(A, w))
```
Then you would generate B;A;C instead of A;B;C. If A is influenced by state that is changed by B, then this is wrong.
Essentially, 
```
seqn(
  for {
   _ <- write(S_1)
   _ <- write(S_2)
   ...
   - <- write(S_N)
  } yield Z
)
```
is the same as ``unit(vpr.Seqn(S_1, S_2, ... , S_N, Z))`. In an earlier version, I had more complicated writer that enforced this. But dealing with them was a pain, so I reduced it to the two writer that we have now. However, as a rule of thump, for the encoding of statements, if you have a 'write' or 'assert' (anything that generates statements), then you should put a seqn arround.

Reply:
Thank you for a thorough explanation!